### DISCUSSION TITLE
About additional pages

### DISCUSSION THREAD

**Discussion by zRains** [vuepress/core#69] [Q&A] [answered]
Is there any configuration option like `addpitionalPages`  in vuepress1. x ? or new routes can only be set through the clientappenhance file ?

**Comment by yzx9** [comment, marked as answer]
Upd: <https://vuepress2.netlify.app/advanced/cookbook/adding-extra-pages.html#adding-extra-pages>

I ran into the same problem. Currently, I think there is not a particular API for creating additional page. So, I use `onInitialized` hook to achieve it. However, I can't create a homepage that has a special path: `{ path: '/' }`. It works well in dev mode, but take a bug in buiding mode. 

```typescript
import { createPage, PageOptions } from "@vuepress/core"

export const theme: Theme = (themeOptions, app) => {
  return {
    onInitialized : async (app: App) => {
      const option: PageOptions = {
        path: "/about.html",
        frontmatter: {
          layout: "About",
        },
      }
      const page = await createPage(app, option)
      app.pages.push(page)
    }
  }
}
```

**Reply by zRains** [reply]
Good idea! I'll try your solution. I once tried to [add dynamic routes](https://stackoverflow.com/questions/58118372/vuepress-dynamic-routes-and-render-as-page) through vue-router, but after the refresh operation, it was 404:
``` ts
import { ClientAppEnhance } from '@vuepress/client'
import Guider from './layouts/Guider.vue'

const clientAppEnhance: ClientAppEnhance = ({ router}) => {
  router.addRoute({
    path: '/guide',
    component: Guider,
  })
}
```
Now I can only explicitly create a route `/guide` by creating an empty .md file:
``` bash
├─.vuepress
├─blog
│      editor.md
└─guide
        README.md
```
It works well in dev and buiding mode, but I feel like I've created unnecessary files.

**Reply by meteorlxy** [reply]
Dynamic routes should work after https://github.com/vuepress/vuepress-next/issues/100
